web_src/js/features/common-global.js

```diff
@@ -313,8 +313,22 @@ export function initGlobalButtons() {
     alert('Nothing to hide');
   });
 
-  $('.show-modal.button').on('click', function () {
-    $($(this).data('modal')).modal('show');
+  $('.show-modal').on('click', function () {
+    const modalDiv = $($(this).data('modal'));
+    for (let i = 0; i < this.attributes.length; i++) {
+      const attrib = this.attributes[i];
+      if (!attrib.name.startsWith('data-modal-')) {
+        continue;
+      }
+      const id = attrib.name.substr(11);
+      const target = modalDiv.find(`#${id}`);
+      if (target.is('input')) {
+        target.val(attrib.value);
+      } else {
+        target.text(attrib.value);
+      }
+    }
```

Comment:
```suggestion
for (const attrib of this.attributes) {
  if (!attrib.name.startsWith('data-modal-')) {
    continue;
  }
  const id = attrib.name.substr(11);
  const target = modalDiv.find(`#${id}`);
  if (target.is('input')) {
    target.val(attrib.value);
  } else {
    target.text(attrib.value);
  }
```

Please test this, I'm not sure if `.attributes` can be iterated like this. If not, `Array.from(this.attributes)` should work.